### PR TITLE
Issue #28: docker-compose, github action, and one test.

### LINKED ISSUE #28: docker-compose setup for Collabora?
For testing and demo purposes, it would be useful for this module to ship with a docker-compose setup (or .ddev).

This would contain:
- A web container and a database container for Drupal.
- A container for a local collabora instance.

@hfiguiere Is there an example docker-compose.yml out there that I could look at?

I found this, which is from 2021: https://techoverflow.net/2021/08/19/how-to-run-collabora-office-for-nextcloud-using-docker-compose/

_For now, let's treat this issue just as a support request. Currently this change is not on our agenda but I am sure this topic will become relevant soon._

### PR DESCRIPTION
**Pull request by donquixote**
Fixes #28.

Changes:
- Local Drupal installation with docker-compose, including a Collabora container, so that:
  - Site builders get a local demo to evaluate the module.
  - Contributors can use the local installation to test new functionality.
- A new test case that tests the embedded Collabora editor in readonly mode.
  - This is an ExistingSite* test, using the existing testing/demo installation. It turns out that regular browser tests are difficult to do for this purpose.
- A github action to run all the tests.

### Testing instructions
Follow the updated steps for "local installation" in the README.md.

_If the editor does not open, you may want to cherry-pick the commit from [here](https://github.com/CollaboraOnline/collabora-drupal/commit/371205d13bb7f90fc9344d5e5945c4da5c5c056c) to replace `file_get_contents()` with curl._

To test the "StaleElementReference" problem, you can auto-repeat the test until it occurs.
E.g.
```sh
watch -n1 -e -x docker-compose exec web ./vendor/bin/phpunit -vvv --debug tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php
```
If you remove the last commit, or change the `retryIfStaleElement` to not catch the exception, then the StaleElementReference will appear.

### Known issues
For now, the actions run in the fork repo only.
https://github.com/donquixote/collabora-drupal/actions
I expect this will change once any PR is merged that contains actions.

For now, we use a docker image from `fpfis`, and we rely on `openeuropa/task-runner` to install the demo/testing site.
This follows the same pattern as e.g. https://www.drupal.org/project/description_list_field, which are published on drupal.org, but where the main development happens in github.
We may rethink this in the future.

For now I need to replace the `file_get_contents` by cherry-picking the commit from [here](https://github.com/CollaboraOnline/collabora-drupal/commit/371205d13bb7f90fc9344d5e5945c4da5c5c056c), to make the local install work properly.
I don't want to include the change in this PR.

I tried a version of the matrix with Drupal 11, and with `--prefer-lowest`, but these are failing.
We should do these another time.

### CODE REVIEW

File: demo-config/core.entity_form_display.media.document.default.yml
Comment (donquixote) [inline demo-config/core.entity_form_display.media.document.default.yml:52]: I am not sure what would be the best place for these config files..

File: tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php
Comment (donquixote) [inline tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php:80]: This method with `file_put_contents()` seems fragile for repeated calls in an ExistingSite test, but I think it won't cause any problems that we would care about.

**Comment by donquixote** [comment]
@hfiguiere This needs at least one review from my colleagues, please don't merge before that :)

File: README.md
Comment (hfiguiere) [inline README.md:64]: really we should stick to the github repo as it will always have the latest code. And the releases are tagged anyway

Otherwise the https URL should be used by default:
 
```suggestion
git clone https://git.drupalcode.org/project/collabora_online.git
```

Comment (donquixote) [reply README.md]: I removed the Drupal git url.
If it ever becomes more relevant, we can change the README.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php
Comment (brummbar) [inline tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php:54]: Avoid nesting, it makes understanding which part failed and debugging more difficult.
```
$input = $assert_session-> waitForRequiredElement('css', 'input#document-name-input');
$value = $this->waitForElementValue($text_file_basename . '.txt', $input);
$this->assertNotNull($value, 'The value was not the expected filename.';

File: tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php
Comment (brummbar) [inline tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php:118]: Instead of pinging again the browser to retrieve the element, just use the return value of the previous call.
```
$element = $this->assertSession()->waitForElement($selector, $locator, $timeout);
$this->assertNotNull($element, sprintf('Could not find element with %s, %s.', $selector, $locator);
return $element;

**Review by brummbar — CHANGES_REQUESTED** [review]

File: composer.json
Comment (donquixote) [reply composer.json]: I removed both devel and admin_toolbar.

The reason to add at least admin_toolbar would be convenience for developers, and for people who want to test-run the module to decide if they like it.

File: tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php
Comment (donquixote) [inline tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php:44]:
```suggestion
          return (string) $document_field->getValue() !== '';
```
We could use this to get the old behavior, where we stop waiting if the string is not empty.

Comment (donquixote) [reply tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php:44]: I previously thought we need to repeat the selector every time to avoid the stale reference.
But seems like the opposite is the case.

**Comment by donquixote** [comment]
I removed leftover code and combined _some_ commits.
I am still keeping some of the commits separate which represent a learning process that could be valuable.

**Comment by donquixote** [comment]
@hfiguiere This PR is good to go from our side.

**Comment by hfiguiere** [comment]
marged. thanks!

**Pull request merged**
